internal_worker.go

```diff
@@ -348,43 +348,59 @@ func (th *hostEnvImpl) AddActivityRegistrationInterceptor(i interceptorFn) {
 	}
 }
 
-func (th *hostEnvImpl) RegisterWorkflow(af interface{}) error {
+func (th *hostEnvImpl) RegisterWorkflow(af interface{}, options RegisterWorkflowOptions) error {
 	// Validate that it is a function
 	fnType := reflect.TypeOf(af)
 	if err := validateFnFormat(fnType, true); err != nil {
 		return err
 	}
-	// Check if already registered
-	fnName := getFunctionName(af)
-	if _, ok := th.getWorkflowFn(fnName); ok {
-		return fmt.Errorf("workflow type \"%v\" is already registered", fnName)
+	names := make([]string, 0, 2)
+	names = append(names, getFunctionName(af))
```

Comment:
Based on offline discussion, this behavior will stay. Register will implicitly add the fully qualified function name and user will not need to call Register twice.